packages/swingset-xsnap-supervisor/README.md

```diff
@@ -0,0 +1,56 @@
+
+When the Endo source bundler is used, with `{ moduleFormat: 'endoZipBase64' }`, the generated bundles include package names and version numbers of all the source files in the transitive import graph starting from the entry point. This means the bundle contents are sensitive to version-number changes, even if the code itself remains the same.
+
+The supervisor bundle does not use this format: the initial `xsnap` environment does not yet have a loader for `endoZipBase64`-format bundles, so instead we use `nestedEvaluate`, which is fairly easy to evaluate into place. This format does not include as much metadata (like version numbers), nevertheless using a separate package for the bundle makes it easier to maintain stability of the contents.
+
+## API
+
+Bundles are JS Objects with a `moduleFormat` property (a string), and then a few other properties that depend on the module format. The `nestedEvaluate` format includes a `.source` property (a large string, suitable for IIFE evaluation) and sometimes a `.sourceMap` property. We can also JSON-serialize the bundle for storage in a single string, on disk or in a database.
+
+The primary job of this package is to provide the supervisor bundle in a form that can be delivered to the `xsnap` process for evaluation in its "Start Compartment". So the primary API is a `getSupervisorBundle()`, which returns the JS Object form.
+
+```js
+import { getSupervisorBundle } from '@agoric/swingset-xs-supervisor';
+const bundle = await getSupervisorBundle();
+
+assert.equal(bundle.moduleFormat, 'nestedEvaluate');
+await worker.evaluate(`(${bundle.source}\n)()`.trim());
+```
+
+To help detect version drift or build/import problems, the package also exports the hex-encoded SHA256 hash of the JSON-stringified bundle. This should be identical to running `/usr/bin/shasum -a 256` on the pathname recorded in the internal `bundlePaths.supervisor` (typically `swingset-xs-supervisor/bundles/supervisor.bundle`). Clients which have selected a particular version of `@agoric/swingset-xs-supervisor` in their `package.json` can retrieve this hash at import time and compare it against a hard-coded copy, to assert that their preferred version is actually in use. Such clients would need to update their copy of the hash each time they deliberately switch to a different version.
+
+```js
+import { supervisorBundleSHA256 } from '@agoric/swingset-xs-supervisor';
+const expected = '54434e4a0eb0c2883e30cc43a30ac66bb792bec3b4975bb147cb8f25c2c6365a';
+assert.equal(supervisorBundleSHA256, expected, 'somehow got wrong version');
+```
+## Bundle Generation
+
+If you have a source tree, you must run `yarn build` to generate the bundles, before this package can be used.
+
+The files in `src/` are visible to downstream users of this package, and they implement the API described above.  The files in `lib/` are used to create the bundle during `yarn build`, and are *not* directly visible to the downstream users of this package.
```

Comment:
Nit: this is actually the opposite of the old node package pattern, where `src` is the raw source, usually not usable outside the package, and `lib` is the public API.

Reply:
If we restrict deep imports it doesn't really matter